cmake/macros/macro_populate_target_properties.cmake

```
@@ -126,4 +126,14 @@ function(populate_target_properties _target _build)
     ${DEAL_II_LIBRARIES} ${DEAL_II_LIBRARIES_${_build}}
     ${DEAL_II_TARGETS} ${DEAL_II_TARGETS_${_build}}
     )
+
+
+   # For release builds (and their corresponding object files),
+   # use interprocedural optimizations if possible
+   if (("${_build}" STREQUAL "RELEASE") AND ("${DEAL_II_COMPILER_SUPPORTS_IPO}" STREQUAL "YES"))
```

Don't use a `STREQUAL` here, because this is a boolean and could also read `TRUE`, `true`, `on`, `ON`, `yEs`, etc.:
```suggestion
   if (DEAL_II_USE_LTO AND ("${_build}" STREQUAL "RELEASE"))
```